### PR TITLE
Moves potential actions creation into separate class

### PR DESCRIPTION
**Pull request by damianszczepanik**

### CODE REVIEW

File: src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java [hunk 1]
```diff
@@ -444,17 +440,6 @@ private void addScmDetails(List<Fact> factsList) {
         }
     }
 
-    private void addPotentialAction(Card card, List<Fact> factsList) {
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java]: ``addPotentialAction`` has been moved to separate class

File: src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java [hunk 2]
```diff
@@ -476,37 +461,4 @@ private String evaluateMacro(String template) {
             throw new IllegalArgumentException(e);
         }
     }
-
-    private void pullRequestActionable(List<PotentialAction> paList, List<Fact> factsList) {
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/Office365ConnectorWebhookNotifier.java]: as well as ``pullRequestActionable``

File: src/main/java/jenkins/plugins/office365connector/PotentialActionBuilder.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    public List<PotentialAction> buildPotentialAction(List<Fact> factsList) {
+        List<PotentialAction> potentialActions = pullRequestActionable(factsList);
+
+        potentialActions.add(buildViewBuild());
+        potentialActions.addAll(pullRequestActionable(factsList));
+
+        return potentialActions;
+    }
+
+    private PotentialAction buildViewBuild() {
```
Comment (damianszczepanik) [inline src/main/java/jenkins/plugins/office365connector/PotentialActionBuilder.java]: ... and split into smaller method which will be easier to test later.

Comment (casz) [reply src/main/java/jenkins/plugins/office365connector/PotentialActionBuilder.java]: Can we please rename this method 😞 buildActionable?

Comment (damianszczepanik) [reply src/main/java/jenkins/plugins/office365connector/PotentialActionBuilder.java]: This probably not as this is ``private`` and it produces one concrete action but the one above which is ``public`` for sure.

But then maybe rename this class to ``ActionableBuilder`` - what do you think?

**Comment by damianszczepanik** [comment]
Will resolve conflicts and apply FactsBuilder which is now merged

**Comment by outconn** [comment]
please resolve the conflicts

File: src/main/java/jenkins/plugins/office365connector/ActionableBuilder.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            }
+            ContributorMetadataAction cma = job.getAction(ContributorMetadataAction.class);
+            if (cma != null) {
+                String contributor = cma.getContributor();
+                String contributorDisplayName = cma.getContributorDisplayName();
+                String author = StringUtils.defaultIfBlank(cma.getContributor(), cma.getContributorDisplayName());
+                if (StringUtils.isNotBlank(contributor) && StringUtils.isNotBlank(contributorDisplayName))
+                    author = String.format("%s (%s)", cma.getContributor(), cma.getContributorDisplayName());
+
+                if (StringUtils.isNotBlank(author))
```
Comment (casz) [inline src/main/java/jenkins/plugins/office365connector/ActionableBuilder.java]: This line is not necessary with the new facts Builder :)

**Pull request merged**